src/Xamarin.Android.Build.Tasks/Tasks/JavaToolTask.cs

```diff
@@ -181,13 +201,11 @@ protected override void LogEventsFromTextOutput (string singleLine, MessageImpor
 			var match = CodeErrorRegEx.Match (singleLine);
 			var exceptionMatch = ExceptionRegEx.Match (singleLine);
 
-			if (match.Success || exceptionMatch.Success) {
-				Log.LogMessage (MessageImportance.High, singleLine);
+			if (match.Success || exceptionMatch.Success && !foundError) {
```

Comment:
What also concerns me is "context" and "do the tools we execute behave 'sanely'?".

Here we are assuming that the *only* messages to "forward to the user" will be either:

 1. Messages for which `CodeErrorRegEx` or `ExceptionRegEx` match, or
 2. All messages which after (1)

This suggests that if any "useful information" *precedes* the first "error" message, it will be *skipped* and never reported.

What I would prefer would be to record *every* line, status & error alike, and *when an error is found*, we dump *all* of the lines.

Suggesting instead:

```csharp
partial class JavaToolTask {

    List<string> lines = new();

    protected override void LogEventsFromTextOutput (string singleLine, MessageImportance messageImportance)
    {
        lines.Add (singleLine);
        base.LogEventsFromTextOutput (singleLine, messageImportance);  // not sure why/when we would skip this?

        if (foundError) {
            return;
        }
        var match = CodeErrorRegEx.Match (singleLine);
        var exceptionMatch = ExceptionRegEx.Match (singleLine);
        foundError = match.Success || exceptionMatch.Success;
    }
}
```